File: gitdot-api/src/endpoint/organization/add_member.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::OrganizationMemberResource};

pub struct AddMember;

impl Endpoint for AddMember {
    const PATH: &'static str = "/organization/{org_name}/repositories";
    const METHOD: http::Method = http::Method::POST;

    type Request = AddMemberRequest;
    type Response = AddMemberResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct AddMemberRequest {
    pub user_name: String,
    pub role: String,
    pub role_description: Option<String>,
}

pub type AddMemberResponse = OrganizationMemberResource;

File: gitdot-web/packages/gitdot-api/src/endpoint/organization/add-member.ts
import { z } from "zod";
import { OrganizationMemberResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const AddMemberRequest = z.object({
  user_name: z.string(),
  role: z.string(),
  role_description: z.string().nullable().optional(),
});
export type AddMemberRequest = z.infer<typeof AddMemberRequest>;

export const AddMemberResponse = OrganizationMemberResource;
export type AddMemberResponse = z.infer<typeof AddMemberResponse>;

export const AddMember = {
  path: "/organization/{org_name}/repositories",
  method: "POST",
  request: AddMemberRequest,
  response: AddMemberResponse,
} as const satisfies Endpoint;
export type AddMember = typeof AddMember;
